### DISCUSSION TITLE
Pimcore 10 Backend UI | Overriding many-to-many search dialog

### DISCUSSION THREAD

**Discussion by QWpreischl** [pimcore/pimcore#11235] [Q&A / Support]
Hello Team,

i am trying to add a textarea field to the search fields of the many-to-many object relation search right before the search button, that is defined at
/vendor/pimcore/pimcore/bundles/AdminBundle/Resources/public/js/pimcore/element/selector/object.js

I have a created a startup.js with a preOpenObject hook, that works fine (executes my test console.log).
In the startup.js i've tried adding the Textarea field to the getForm function by using the Ext.override function to override the pimcore.element.selector.object class like this:

```
Ext.override('pimcore.element.selector.object', {
            getForm: function() {
                /* ... content of original function ... */
                
                compositeConfig.items.push({
                    xtype: "textareafield",
                    anchor: '100%',
                    height: 100,
                    name: "textfieldName",
                    editable: true,
                    emptyText: "123, 456, 789, ...",
                    hideLabel: true
                });

                /* ... content of original function ... */
            }
        });
```

This doesn't work tho and i haven't found a solution that works inside of the search dialog popup yet since the content of it gets loaded dynamically. 
Adding the textareafield to the object.js file displays the field correctly, but thats not an option since it will be overwritten from pimcore updates.

So how can i add the field to the search dialog without losing the ability to update the system?

**Comment by abdelrhmanyosry** [comment]
did you find any solution for it?

**Comment by thomasschradermasnou** [comment]
Try this:

```js
pimcore.element.selector.object.property.getForm = function() {
 /* ... content of original function ... */
                
                compositeConfig.items.push({
                    xtype: "textareafield",
                    anchor: '100%',
                    height: 100,
                    name: "textfieldName",
                    editable: true,
                    emptyText: "123, 456, 789, ...",
                    hideLabel: true
                });

                /* ... content of original function ... */
            }
}
```
